### DISCUSSION TITLE
Adonis Relationship: manyToManyThrough?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#3623] [Help]
There is a relationship called hasManyThrough. But does it exist manyToManyThrough?

In my app, companies can belong to multiple lists and a team can have multiple lists.

Company <=> Company_List <=> List <=> Team

I would like to retrieve paginated results of companies that belong to a specific team

**Comment by thetutlage** [comment]
There is no `manyToManyThrough` right now. Maybe you can work-around it as follows.

- Get id of lists that belongs to a team
- Get companies `whereIn(listIds)`.

**Comment by cawa-93** [comment]
I have a similar use case. Each `User` have `role` column and `belongsTo` `Role`. Each `Role` have `manyToMany` relationship with `capabilities`. 

```mermaid
classDiagram
direction RL
class capabilities {
   varchar(20) id
}
class capability_role {
   varchar(20) role_id
   varchar(20) capability_id
   integer id
}
class roles {
   varchar(20) id
}
class users {
   varchar(255) email
   varchar(180) password
   varchar(255) remember_me_token
   varchar(20) role
   integer id
}

capabilities  -->  capability_role
roles  -->  capability_role
roles  -->  users
```

I don't fully understand whether it's possible to use `HasManyThrough` to associate a `User` with `Capability` through `Role`

**What I want:**
```ts
await auth.user.load('capabilities')
auth.user.capabilities // [{id: 'cap1'}, {id: 'cap2'}]
```

**What I actually should do:**
```ts
await auth.user.load('role')
await auth.user.role.load('capabilities')
// or
await auth.user.load('role', (q) => q.preload('capabilities'))

auth.user.role.capabilities // [{id: 'cap1'}, {id: 'cap2'}]
